app/lib/search/index_ducene.dart

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:ducene/analysis.dart';
+import 'package:ducene/document.dart';
+import 'package:ducene/search.dart';
+import 'package:ducene/util.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+
+import '../shared/configuration.dart';
+import '../shared/search_service.dart';
+
+/// The [PackageIndex] registered in the current service scope.
+PackageIndex get packageIndex => ss.lookup(#_ducene_package_index);
+
+/// Register a new [PackageIndex] in the current service scope.
+void registerPackageIndex(PackageIndex index) =>
```

Comment:
For the other things we use the service scope for we postfix the name with service. Maybe we should also do that here: `packageIndexService`.